phpBB/phpbb/notification/manager.php

```diff
@@ -275,10 +275,9 @@ public function mark_notifications_read($notification_type_name, $item_id, $user
 		$sql = 'UPDATE ' . $this->notifications_table . "
 			SET notification_read = 1
 			WHERE notification_time <= " . (int) $time .
-				(($notification_type_name !== false) ? ' AND ' .
-					(is_array($notification_type_name) ? $this->db->sql_in_set('notification_type_id', $this->get_notification_type_ids($notification_type_name)) : 'notification_type_id = ' . $this->get_notification_type_id($notification_type_name)) : '') .
-				(($user_id !== false) ? ' AND ' . (is_array($user_id) ? $this->db->sql_in_set('user_id', $user_id) : 'user_id = ' . (int) $user_id) : '') .
-				(($item_id !== false) ? ' AND ' . (is_array($item_id) ? $this->db->sql_in_set('item_id', $item_id) : 'item_id = ' . (int) $item_id) : '');
+				(($notification_type_name !== false) ? ' AND ' . $this->db->sql_in_set('notification_type_id', $this->get_notification_type_ids($notification_type_name)) : '') .
```

Comment:
Sure, shouldn't add any BC breaks either. Just simply remove the array constraint and do something like:
```
if (!is_array($notification_type_names))
{
    $notification_type_names = array($notification_type_names);
}
```